meta-mender-tegra/recipes-bsp/u-boot/patches/0010-tegra-mender-auto-configured-modified.patch

```diff
@@ -1,80 +1,165 @@
+index af2ecae0bd..d78aab1057 100644
+--- a/configs/p2771-0000-500_defconfig
++++ b/configs/p2771-0000-500_defconfig
+@@ -1,8 +1,8 @@
+ CONFIG_ARM=y
+ CONFIG_ARCH_TEGRA=y
+ CONFIG_SYS_TEXT_BASE=0x80080000
+-CONFIG_ENV_SIZE=0x8000
+-CONFIG_ENV_OFFSET=0x3D8000
++CONFIG_ENV_SIZE=0x20000
++CONFIG_ENV_OFFSET=0x3bb000
+ CONFIG_NR_DRAM_BANKS=1043
+ CONFIG_TEGRA186=y
+ CONFIG_OF_BOARD_SETUP=y
+diff --git a/configs/p3450-0000_defconfig b/configs/p3450-0000_defconfig
+index e5cf80fa0d..397ef7d39b 100644
+--- a/configs/p3450-0000_defconfig
++++ b/configs/p3450-0000_defconfig
+@@ -1,8 +1,8 @@
+ CONFIG_ARM=y
+ CONFIG_ARCH_TEGRA=y
+ CONFIG_SYS_TEXT_BASE=0x80080000
+-CONFIG_ENV_SIZE=0x8000
+-CONFIG_ENV_OFFSET=0x3D8000
++CONFIG_ENV_SIZE=0x20000
++CONFIG_ENV_OFFSET=0x3b0000
+ CONFIG_ENV_SECT_SIZE=0x1000
+ CONFIG_NR_DRAM_BANKS=16
+ CONFIG_TEGRA210=y
+diff --git a/configs/p3450-0002_defconfig b/configs/p3450-0002_defconfig
+index 55c2957c33..52aba36cae 100644
+--- a/configs/p3450-0002_defconfig
++++ b/configs/p3450-0002_defconfig
+@@ -50,7 +50,7 @@ CONFIG_USB_GADGET_DOWNLOAD=y
+ CONFIG_USB_HOST_ETHER=y
+ CONFIG_USB_ETHER_ASIX=y
+ CONFIG_ENV_IS_IN_MMC=y
+-CONFIG_ENV_SIZE=0x8000
++CONFIG_ENV_SIZE=0x20000
+ CONFIG_ENV_SECT_SIZE=0x1000
+-CONFIG_ENV_OFFSET=0x3D8000
++CONFIG_ENV_OFFSET=0x3bb000
+ CONFIG_BOOTP_PREFER_SERVERIP=y
 diff --git a/include/config_defaults.h b/include/config_defaults.h
-index ad08c1d335..3dad911845 100644
+index d211a41ef2..5e7be1ac54 100644
 --- a/include/config_defaults.h
 +++ b/include/config_defaults.h
-@@ -21,3 +21,5 @@
- #define CONFIG_PARTITIONS 1
+@@ -10,3 +10,5 @@
+ #define _CONFIG_DEFAULTS_H_
  
  #endif
 +#define CONFIG_BOOTCOUNT_LIMIT
 +#define CONFIG_BOOTCOUNT_ENV
 diff --git a/include/config_distro_bootcmd.h b/include/config_distro_bootcmd.h
-index d7182244cb..32df04aef1 100644
+index fc0935fa21..7433b0b760 100644
 --- a/include/config_distro_bootcmd.h
 +++ b/include/config_distro_bootcmd.h
-@@ -397,7 +397,6 @@
+@@ -486,7 +486,6 @@
  		"done\0"
  
  #ifndef CONFIG_BOOTCOMMAND
 -#define CONFIG_BOOTCOMMAND "run distro_bootcmd"
  #endif
  
  #endif  /* _CONFIG_CMD_DISTRO_BOOTCMD_H */
-diff --git a/include/configs/p2771-0000.h b/include/configs/p2771-0000.h
-index 61310fbdf2..3010e2f5d4 100644
---- a/include/configs/p2771-0000.h
-+++ b/include/configs/p2771-0000.h
-@@ -26,7 +26,6 @@
- #define CONFIG_ENV_IS_IN_MMC
- #define CONFIG_SYS_MMC_ENV_DEV		0
- #define CONFIG_SYS_MMC_ENV_PART		2
--#define CONFIG_ENV_OFFSET		(-CONFIG_ENV_SIZE)
- 
- /* PCI host support */
- #define CONFIG_PCI
-diff --git a/include/configs/tegra-common.h b/include/configs/tegra-common.h
-index 4c4a1ea1c6..0446ba594b 100644
---- a/include/configs/tegra-common.h
-+++ b/include/configs/tegra-common.h
-@@ -34,7 +34,6 @@
- 
- /* Environment */
- #define CONFIG_ENV_VARS_UBOOT_CONFIG
--#define CONFIG_ENV_SIZE			0x2000	/* Total Size Environment */
- 
- /*
-  * NS16550 Configuration
 diff --git a/include/configs/tegra186-common.h b/include/configs/tegra186-common.h
```

Comment:
Rather than trying to pick all this apart and put it back together, I put in some time and managed to completely regenerate this from the current mender auto-patching scripts. In retrospect I should have done this from the start because it really simplifies the changes. All that remains are config changes and removing a couple environment device selection definitions removed from the board header files.